dex/testing/loadbot/loadbot.go

```diff
@@ -0,0 +1,461 @@
+	epochDuration              uint64
+)
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+func rpcAddr(symbol, node string) string {
+	switch symbol {
+	case dcr:
+		switch node {
+		case alpha:
+			return alphaCfgDCR["rpclisten"]
+		case beta:
+			return betaCfgDCR["rpclisten"]
+		}
+	case btc:
+		switch node {
+		case alpha:
+			return alphaCfgBTC["rpcport"]
+		case beta:
+			return betaCfgBTC["rpcport"]
+		}
+	}
+	panic("rpcAddr: unknown symbol-node combo " + symbol + "-" + node)
+}
+
+func returnAddress(symbol, node string) string {
+	switch symbol {
+	case dcr:
+		switch node {
+		case alpha:
+			return alphaAddrDCR
+		case beta:
+			return betaAddrDCR
+		}
+	case btc:
+		switch node {
+		case alpha:
+			return alphaAddrBTC
+		case beta:
+			return betaAddrBTC
+		}
+	}
+	panic("returnAddress: unknown symbol-node combo " + symbol + "-" + node)
+}
+
+type botWallet struct {
+	form     *core.WalletForm
+	name     string
+	node     string
+	symbol   string
+	pass     []byte
+	assetID  uint32
+	minFunds uint64
+	maxFunds uint64
+	address  string
+	numCoins int
+}
+
+func newBotWallet(symbol, node, name string, pass []byte, minFunds, maxFunds uint64, numCoins int) *botWallet {
+	var form *core.WalletForm
+	switch symbol {
+	case dcr:
+		form = &core.WalletForm{
+			AssetID: dcrID,
+			Config: map[string]string{
+				"account":   name,
+				"username":  "user",
+				"password":  "pass",
+				"rpccert":   filepath.Join(dextestDir, "dcr/"+node+"/rpc.cert"),
+				"rpclisten": rpcAddr(symbol, node),
+			},
+		}
+	case btc:
+		form = &core.WalletForm{
+			AssetID: btcID,
+			Config: map[string]string{
+				"walletname":  name,
+				"rpcuser":     "user",
+				"rpcpassword": "pass",
+				"rpcport":     rpcAddr(symbol, node),
+			},
+		}
+	}
+	return &botWallet{
+		form:     form,
+		name:     name,
+		node:     node,
+		symbol:   symbol,
+		pass:     pass,
+		assetID:  form.AssetID,
+		minFunds: minFunds,
+		maxFunds: maxFunds,
+		numCoins: numCoins,
+	}
+}
+
+// mineAlpha will mine a single block on the alpha node of the asset indicated
+// by symbol.
+func mineAlpha(symbol string) {
+	harnessCtl(symbol, "./mine-alpha 1")
+}
+
+// mineBeta will mine a single block on the beta node of the asset indicated
+// by symbol.
+func mineBeta(symbol string) {
+	harnessCtl(symbol, "./mine-beta 1")
+}
+
+type harnessResult struct {
+	err    error
+	output string
+	cmd    string
+}
+
+func (res *harnessResult) String() string {
+	if res.err != nil {
+		return fmt.Sprintf("error running harnes command %q: %v", res.cmd, res.err)
+	}
+	return fmt.Sprintf("response from harness command %q: %v", res.cmd, res.err)
+}
+
+// harnessCtl will run the command from the harness-ctl directory for the
+// specified symbol.
+func harnessCtl(symbol, cmd string, args ...interface{}) <-chan *harnessResult {
+	cmd = fmt.Sprintf(cmd, args...)
+	harnessCtl := filepath.Join(dextestDir, symbol, "harness-ctl")
+	compoundCmd := fmt.Sprintf("cd %s; %s", harnessCtl, cmd)
```

Comment:
@itswisdomagain, you might find this handy. If you want the harness-ctl calls to be synchronous, just run the scripts directly instead of through tmux.